tritium-lib/src/main/java/com/palantir/tritium/proxy/ByteBuddyInstrumentationAdvice.java

```diff
@@ -52,53 +49,20 @@ static InvocationContext enter(
             @Advice.FieldValue("instrumentationFilter") InstrumentationFilter filter,
             @Advice.FieldValue("invocationEventHandler") InvocationEventHandler<?> eventHandler,
             @Advice.FieldValue("methods") Method[] methods,
-            @Advice.FieldValue("log") Logger logger,
-            @Advice.FieldValue("DISABLED_HANDLER_SENTINEL") InvocationContext disabledHandlerSentinel,
             @MethodIndex int index) {
-        Method method = methods[index];
-        try {
-            if (eventHandler.isEnabled() && filter.shouldInstrument(proxy, method, arguments)) {
-                return eventHandler.preInvocation(proxy, method, arguments);
-            }
-            return disabledHandlerSentinel;
-        } catch (RuntimeException | Error t) {
-            if (logger.isWarnEnabled()) {
-                logger.warn(
-                        "Failure occurred handling 'preInvocation' invocation on: {}",
-                        UnsafeArg.of("instance", Objects.toString(proxy)),
-                        t);
-            }
-            return null;
-        }
+        return Handlers.preWithEnabledCheck(eventHandler, filter, proxy, methods[index], arguments);
```

Comment:
This may have a substantial (positive) impact on byte-buddy bytecode generation performance. Advice methods are inlined into each instrumented method, this change shares additional code for better reuse and less resulting bytecode.

Reply:
👍 on extracting `Handlers`